From 7d4fd981e6e1aa463d3f309483c10fcf068d4f7a Mon Sep 17 00:00:00 2001
From: Alexandru Ardelean <alexandru.ardelean@analog.com>
Date: Wed, 30 Oct 2019 09:35:00 +0200
Subject: [PATCH 0680/1851] dmaengine: xilinx: dpdma: fix implicit fall-through
 statements

commit e562d0d58f9257d839fa63131df60cbb4b184ae9 from
https://github.com/Xilinx/linux-xlnx.git

Newer kernel versions have this warning enabled. This change adds the
necessary `fall-through` comments to suppress them.

Signed-off-by: Alexandru Ardelean <alexandru.ardelean@analog.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/dma/xilinx/xilinx_dpdma.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/dma/xilinx/xilinx_dpdma.c b/drivers/dma/xilinx/xilinx_dpdma.c
index 89ade893f51c..cf17c20085e6 100644
--- a/drivers/dma/xilinx/xilinx_dpdma.c
+++ b/drivers/dma/xilinx/xilinx_dpdma.c
@@ -1744,7 +1744,9 @@ static void xilinx_dpdma_chan_handle_err(struct xilinx_dpdma_chan *chan)
 	switch (chan->active_desc->status) {
 	case ERRORED:
 		dev_dbg(dev, "repeated error on desc\n");
+		/* fall-through */
 	case ACTIVE:
+		/* fall-through */
 	case PREPARED:
 		/* Reschedule if there's no new descriptor */
 		if (!chan->pending_desc && !chan->submitted_desc) {
-- 
2.31.1

